core/src/main/java/com/expediagroup/streamplatform/streamregistry/core/services/ProducerService.java

```diff
@@ -92,7 +97,25 @@
 
   @PreAuthorize("hasPermission(#producer, 'DELETE')")
   public void delete(Producer producer) {
-    throw new UnsupportedOperationException();
+    val existing = unsecuredGet(producer.getKey());
+    if (!existing.isPresent()) {
+      throw new ValidationException("Can't delete " + producer.getKey().getName() + " because it doesn't exist");
+    }
+    handlerService.handleDelete(producer);
+    producerBindingService.findAllAndDelete(producer.getKey());
+    producerRepository.delete(producer);
+  }
+
+  @PreAuthorize("returnObject.isPresent() ? hasPermission(returnObject, 'DELETE') : true")
+  public void findAllAndDelete(StreamKey key) {
+    val example = new Producer(new ProducerKey(
+            key.getDomain(),
+            key.getName(),
+            key.getVersion(),
+            null,
+            null
+    ), null, null);
+    producerRepository.findAll(example).forEach(this::delete);
```

Comment:
I'm not sure this method is implemented for Producers or Consumers. There's a comment in the `DefaultRepository` that says:

```
//This is only used by ProducerBinding and ConsumerBinding
```

Reply:
If this is the case, we probably need to improve our integration tests so that'll pick this up. I'm thinking we might need some tests that check the cascading nature of deletes and ensure that all entities are successfully deleted

Reply:
removed use of deprecated method